src/components/Scheme/SchemeContainer.tsx

```diff
@@ -10,26 +11,50 @@ import { Link, Route, RouteComponentProps, Switch } from "react-router-dom";
 import * as Sticky from "react-stickynode";
 import { from } from "rxjs";
 import { concatMap } from "rxjs/operators";
+import { showNotification } from "reducers/notifications";
+import { IRootState } from "reducers";
+import { connect } from "react-redux";
 import SchemeInfoPage from "./SchemeInfoPage";
 import SchemeProposalsPage from "./SchemeProposalsPage";
 import * as css from "./Scheme.scss";
 
+interface IDispatchProps {
+  showNotification: typeof showNotification;
+}
 
-interface IProps {
+interface IExternalProps {
   currentAccountAddress: Address;
+  history: H.History;
 }
 
-export default class SchemeContainer extends React.Component<IProps & RouteComponentProps<any>, null> {
+type IProps = IExternalProps & IDispatchProps;
+
+const mapStateToProps = (_state: IRootState, ownProps: IExternalProps): IExternalProps => {
+  return ownProps;
+};
+
+const mapDispatchToProps = {
+  showNotification,
+};
+
+class SchemeContainer extends React.Component<IProps & RouteComponentProps<any>, IExternalProps> {
+
+  private async handleNewProposal(daoAvatarAddress: Address, schemeId: any): Promise<void> {
+    if ((await checkWeb3ProviderAndWarn(this.props.showNotification.bind(this)))) {
+      this.props.history.push(`/dao/${daoAvatarAddress}/scheme/${schemeId}/proposals/create`);
+    }
+  }
 
-  public render() {
+  public render(): any {
     const { currentAccountAddress, match } = this.props;
     const schemeId = match.params.schemeId;
     const daoAvatarAddress = match.params.daoAvatarAddress;
+    const _handleNewProposal = (e: any): void => { this.handleNewProposal(daoAvatarAddress, schemeId); e.preventDefault(); /* e.stopPropagation(); */};
```

Comment:
can we break this onto multiple lines

Reply:
also why do we have this? cant you put the preventDefault above?

Reply:
I modified to put it all together in one function.